docs/api.rst

```diff
@@ -1,115 +1,90 @@
 API Reference
 =============
 
-.. module:: attr
 .. module:: attrs
 
-.. currentmodule:: attr
+*attrs* works by decorating a class using `attrs.define` or `attr.s` and then defining attributes on the class using `attrs.field`, `attr.ib`, or type annotations.
 
-*attrs* works by decorating a class using `attrs.define` or `attr.s` and then optionally defining attributes on the class using `attrs.field`, `attr.ib`, or a type annotation.
+What follows is the API explanation, if you'd like a more hands-on tutorial, have a look at `examples`.
 
-If you're confused by the many names, please check out `names` for clarification.
+If you're confused by the many names, please check out `names` for clarification, but the `TL;DR <https://en.wikipedia.org/wiki/TL;DR>`_ is that as of version 21.3.0, *attrs* consists of **two** top-level package names:
 
-What follows is the API explanation, if you'd like a more hands-on introduction, have a look at `examples`.
-
-As of version 21.3.0, *attrs* consists of **two** top-level package names:
-
-- The classic ``attr`` that powered the venerable `attr.s` and `attr.ib`
-- The modern ``attrs`` that only contains most modern APIs and relies on `attrs.define` and `attrs.field` to define your classes.
+- The classic ``attr`` that powers the venerable `attr.s` and `attr.ib`.
+- The newer ``attrs`` that only contains most modern APIs and relies on `attrs.define` and `attrs.field` to define your classes.
   Additionally it offers some ``attr`` APIs with nicer defaults (e.g. `attrs.asdict`).
 
-The ``attrs`` namespace is built *on top of* ``attr`` which will *never* go away.
+The ``attrs`` namespace is built *on top of* ``attr`` -- which will *never* go away -- and is just as stable, since it doesn't constitute a rewrite.
+To keep repetition low and this document at a reasonable size, the ``attr`` namespace is `documented on a separate page <api-attr>`, though.
 
 
 Core
 ----
 
-.. note::
-
-  Please note that the ``attrs`` namespace has been added in version 21.3.0.
-  Most of the objects are simply re-imported from ``attr``.
-  Therefore if a class, method, or function claims that it has been added in an older version, it is only available in the ``attr`` namespace.
-
 .. autodata:: attrs.NOTHING
    :no-value:
 
 .. autofunction:: attrs.define
 
-.. function:: attrs.mutable(same_as_define)
+.. function:: mutable(same_as_define)
```

Comment:
So I was trying to google something I remembered was discussed on the Sphinx tracker, but couldn't recall the exact feature name. I think I've found it now and it's called `:canonical:`. @hynek I think this might be something you're after.
```suggestion
.. function:: mutable(same_as_define)
   :canonical: attrs.define
```

Check out the docs at https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-option-py-function-canonical. FTR I think it's documented quote poorly. I think it was discussed @ https://github.com/sphinx-doc/sphinx/issues/4826 / https://github.com/sphinx-doc/sphinx/pull/9026.

Reply:
That doesn't seem to do anything:

<img width="289" alt="CleanShot 2023-01-05 at 07 44 53@2x" src="https://user-images.githubusercontent.com/41240/210718352-39c7245a-deee-414f-85f0-456e86332ca5.png">

and

```
attrs.mutable                            api.html#attrs.mutable
```

Reply:
Hm… Maybe I've misinterpreted how it works. Let's postpone using it, then.